theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -136,36 +136,22 @@ export class TraceViewerWidget extends ReactWidget {
     }
 
     async initialize(): Promise<void> {
-        /*
-         * TODO: use backend service to find traces
-         */
-
-        const isCancelled = { value: false };
+        const cancellation = new CancellationTokenSource();
 
         // This will show a progress dialog with "Cancel" option
         this.messageService.showProgress({
-            text: 'Open traces',
-            options: {
-                cancelable: true
-            }
-        },
+                text: 'Open traces',
+                options: {
+                    cancelable: true
+                }
+            },
             () => {
-                isCancelled.value = true;
+                cancellation.cancel();
             })
             .then(async progress => {
                 try {
-                    const tracesArray = new Array<Path>();
-                    const fileStat = await this.fileSystem.getFileStat(this.uri.toString());
                     progress.report({ message: 'Finding traces ', work: { done: 10, total: 100 } });
-                    if (fileStat) {
-                        if (fileStat.isDirectory) {
-                            // Find recursivly CTF traces
-                            await this.findTraces(fileStat, tracesArray, isCancelled);
-                        } else {
-                            // Open single trace file
-                            tracesArray.push(this.uri);
-                        }
-                    }
+                    const tracesArray = await this.backendService.findTraces(this.uri.toString(), cancellation.token);
```

Comment:
I pressed cancel while having a breakpoint in the backend service to test the cancellation. In the backend it stopped, but it returned an empty `tracesArray`. So, the check below is then true and I get the error message, `No valid traces found in selected directory ...`. When the array length is 0 and the cancellation was requested I think the error message should not be shown.

Reply:
Moved cancellation check before array size check.